### PR TITLE
feat: Add code compatibility with landing on stars.

### PR DESCRIPTION
**Pull request by Codetoil**
I'm posting this to get feedback.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -3,7 +3,7 @@ org.gradle.jvmargs=-Xmx3G
 # Mod Information
 mod.id=galacticraft-api
 mod.name=GalacticraftAPI
-mod.version=0.4.0-prealpha.23
+mod.version=0.4.0-prealpha.23-tts
```
Comment (Codetoil) [inline gradle.properties:6]: Temporary change to separate my modified API release with the official one.

**Comment by Codetoil** [comment]
@vos6434 It's not finished, but I'd like you or someone else to look over it. I would appreciate some feedback.

File: src/main/java/dev/galacticraft/api/universe/celestialbody/landable/Landable.java [hunk 2]
```diff
@@ -26,43 +26,34 @@
...
+
     /**
-     * Returns the registry key of the {@link Level} this celestial body is linked to
+     * Returns the registry key of the {@link Level} this celestial body is linked to, or {@code null} if it does not exist (say, if it's a decorated planet)
      *
      * @param config the celestial body configuration to be queried
      * @return the registry key of the {@link Level} this celestial body is linked to
      */
-    @NotNull ResourceKey<Level> world(C config);
+    @Nullable ResourceKey<Level> world(C config);
```
Comment (marcus8448) [inline src/main/java/dev/galacticraft/api/universe/celestialbody/landable/Landable.java:49]: If a celestial body is `Landable` you should be able to land on it. If you don't want a landable celestial body, don't use a landable type.

File: src/main/java/dev/galacticraft/api/universe/celestialbody/landable/Landable.java [hunk 3]
```diff
@@ -26,43 +26,34 @@
...
-    int dayTemperature(C config);
-
-    /**
-     * Returns the approximate temperature on this celestial body during the night (in Celsius)
-     *
-     * @param config the celestial body configuration to be queried
-     * @return the approximate temperature on this celestial body during the night
-     */
-    int nightTemperature(C config);
+    @Nullable Integer accessWeight(C config);
```
Comment (marcus8448) [inline src/main/java/dev/galacticraft/api/universe/celestialbody/landable/Landable.java:58]: Just use `-1`, not `null`. I don't see a situation where `-1` would be treated differently than `null`

**Review by marcus8448 — CHANGES_REQUESTED** [review]
The celestial body API is meant to be extensible through the use of interfaces on each type. By making everything inside `Landable` optional, while making all types `Landable` this type based system becomes pointless.

If you want to make a landable star, make your own type that implements `Landable`. Celestial bodies can orbit any type and can be any type; it doesn't need to be the builtin GCAPI ones. (See how `PlanetType` differs from `DecorativePlanet`)

**Comment by Codetoil** [comment]
@marcus8448
If you do that, though, Landable objects has to have be a different class than not-Landable ones, even if they are visibly the same type. (Planets and Decorated Planets, as an example.) It feels a bit wasteful to me, not to mention that it's also a bit hardcode-y. I could change it back if you wish.

**Comment by marcus8448** [comment]
> visibly the same type

What does this mean? They aren't the same type - one you can land on the other you cannot. Since `PlanetType` implements `Landable` it gets treated differently by Galacticraft's code.

> it's also a bit hardcode-y

???
The purpose of this system is to provide a clear divide between celestial bodies with different properties. You are not restricted to GCAPI's builtin types.

> It just doesn't feel right to me

I find it worse to just have a monolithic `CelestialBody` or (a somewhat less monolithic `PlanetType`) class that covers *optional* landing, *optional* orbiting, etc through `null`able methods. End-users/datapack creators could partially implement aspects of celestial bodies and break things.

**Pull request closed without merging**